publication/src/main/scala/hmda/publication/reports/util/ReportUtil.scala

```diff
@@ -44,6 +44,41 @@ object ReportUtil extends SourceUtils {
     Array(i50, i80, i100, i120)
   }
 
+  def larInIncomeInterval(lar: LoanApplicationRegister, applicantIncomeEnum: ApplicantIncomeEnum): Boolean = {
+    val incomeIntervals = calculateMedianIncomeIntervals(lar.geography.msa.toInt)
+    val income = lar.applicant.income.toInt
+    (income < incomeIntervals(0) && applicantIncomeEnum == LessThan50PercentOfMSAMedian) ||
+      (income >= incomeIntervals(0) && income < incomeIntervals(1) && applicantIncomeEnum == Between50And79PercentOfMSAMedian) ||
+      (income >= incomeIntervals(1) && income < incomeIntervals(2) && applicantIncomeEnum == Between80And99PercentOfMSAMedian) ||
+      (income >= incomeIntervals(2) && income < incomeIntervals(3) && applicantIncomeEnum == Between100And119PercentOfMSAMedian) ||
+      (income >= incomeIntervals(3) && applicantIncomeEnum == GreaterThan120PercentOfMSAMedian)
+  }
+
+  def nationalLarsByIncomeInterval(larSource: Source[LoanApplicationRegister, NotUsed]): Map[ApplicantIncomeEnum, Source[LoanApplicationRegister, NotUsed]] = {
```

Comment:
I see what you mean about this operation being really computation-heavy :/. This strategy works for me, afaik; let's just make sure to test out these reports in the cluster to make sure they don't time out.